packages/fluence-js/src/internal/builtins/SingleModuleSrv.ts

```diff
@@ -0,0 +1,125 @@
+import { v4 as uuidv4 } from 'uuid';
+import { SrvDef } from '../_aqua/single-module-srv';
+import { NodeUtilsDef } from '../_aqua/node-utils';
+import { FluencePeer } from '../FluencePeer';
+import { isNode } from 'browser-or-node';
+import { CallParams } from '../commonTypes';
+import { allowOnlyParticleOriginatedAt, SecurityGuard } from './securityGuard';
+
+export const defaultGuard = (peer: FluencePeer) => {
+    return allowOnlyParticleOriginatedAt<any>(peer.getStatus().peerId!);
+};
+
+export class Srv implements SrvDef {
+    private services: Set<string> = new Set();
+
+    constructor(private peer: FluencePeer) {}
+
+    createSecurityGuard: SecurityGuard<'wasm_b64_content'> = defaultGuard(this.peer);
+
+    async create(wasm_b64_content: string, callParams: CallParams<'wasm_b64_content'>) {
+        if (!this.createSecurityGuard(callParams)) {
+            return {
+                success: false,
+                error: 'Security guard validation failed',
+                service_id: null,
```

Comment:
@justprosh @folex please help me understand if we need explicit success in aqua and why

Reply:
That's the limitation of aqua language. No way to represent sum types. At least at the moment

Reply:
I understand there is no way to represent a sum type in aqua, but is `success` really required? Is it possible to understand success as absence of the error

Reply:
it's the limitation of marine and aqua, yes: we don't have good mechanics for errors yet.

`success` is more explicit and easier than checking `error` for emptiness.

besides, it could be that in some code `error` is not empty while `success` is true, or vice versa: result is not empty while `success` is false.

Reply:
this situations that you described are really not cool
non empty error with succes true sound like a bug, as well as vice versa situation